src/applications/model/gaming-traffic-model.cc

```diff
@@ -0,0 +1,104 @@
+/* -*- Mode:C++; c-file-style:"gnu"; indent-tabs-mode:nil; -*- */
+/*
+ * Copyright (c) 2020, University of Padova, Department of Information
+ * Engineering, SIGNET Lab.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation;
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ *
+ * Authors: Salman Mohebi <s.mohebi22@gmail.com>
+ *
+ */
+
+#include "ns3/log.h"
+#include "ns3/simulator.h"
+#include "gaming-traffic-model.h"
+
+namespace ns3 {
+
+  NS_LOG_COMPONENT_DEFINE ("GamingTrafficModel");
+
+  NS_OBJECT_ENSURE_REGISTERED(GamingTrafficModel);
+
+  TypeId
+  GamingTrafficModel::GetTypeId(void) {
+    static TypeId tid = TypeId("ns3::GamingTrafficModel")
+        .SetParent<RandomVariableStream>()
+        .SetGroupName("GamingTraffic")
+        .AddConstructor<GamingTrafficModel>()
+    ;
+    return tid;
+  }
+
+  GamingTrafficModel::GamingTrafficModel() {
+    NS_LOG_FUNCTION (this);
+  }
+
+  double
+  GamingTrafficModel::GetValue(void) {
+    NS_LOG_FUNCTION (this);
+    if (IsValid()) {
+      double thresh = CreateObjectWithAttributes<UniformRandomVariable>()->GetValue();
+      double cumProb = 0;
+      for (int index = 0; index < (int) m_prob.size(); ++index) {
```

Comment:
I don't think you need a cast to int

Reply:
The `m_prob.size()` returns an unsigned type, so to compare I should either cast it to int for use unsigned for index.

Reply:
as stated below, you should use size_t, instead. even better, you could use the [range based for loop](https://en.cppreference.com/w/cpp/language/range-for) as
```cpp
for (double prob : m_prob)
  {
    cumProb += prob;
    //...
  }
```

Reply:
Here I'm iterating over probs, but I need to have access to the indices to return the random variables.
Which is not straight forward using the range-based loop.

Reply:
yes in this case it is true, but in this case use the size_t type